### ISSUE TITLE
loss 全是0

### ISSUE THREAD

**Issue by whk6688** [baichuan-inc/Baichuan2#313]
刚下载的项目
使用官方提供的脚本
hostfile=""
deepspeed --hostfile=$hostfile fine-tune.py  \
    --report_to "none" \
    --data_path "data/belle_chat_ramdon_10k.json" \
    --model_name_or_path "baichuan-inc/Baichuan2-13B-Chat" \
    --output_dir "output" \
    --model_max_length 512 \
    --num_train_epochs 4 \
    --per_device_train_batch_size 16 \
    --gradient_accumulation_steps 1 \
    --save_strategy epoch \
    --learning_rate 2e-5 \
    --lr_scheduler_type constant \
    --adam_beta1 0.9 \
    --adam_beta2 0.98 \
    --adam_epsilon 1e-8 \
    --max_grad_norm 1.0 \
    --weight_decay 1e-4 \
    --warmup_ratio 0.0 \
    --logging_steps 1 \
    --gradient_checkpointing True \
    --deepspeed ds_config.json \
    --bf16 True \
    --tf32 True \
    --use_lora True

结果loss一直为0

![image](https://github.com/baichuan-inc/Baichuan2/assets/17873056/1ddeba78-34e6-4811-a056-5c933ca23622)

**Comment by whk6688** [comment]
另外 Token indices sequence length is longer than the specified maximum sequence length for this model (531 > 512). Running this sequence through the model will result in indexing errors
也有问题吗

**Comment by cristiano-xw** [comment]
我也有这个相同的问题，用的时A100显卡，微调13b-chat

**Comment by prtj** [comment]
pip uninstall xformers 就可以了

**Comment by liunian-Jay** [comment]
同样出现了这个问题

**Comment by gromzhu** [comment]
遇到了同样的问题，pip uninstall xformers解决

**Comment by JackMeiLong** [comment]
pip uninstall xformers，train 的时候又提示没有xformers。

**Comment by joymcg** [comment]
同样的问题，官方有解决方案吗？ 如果没有xformers，就用不上加速的能力了啊

**Comment by shuiyigt** [comment]
modeling_baichuan.py中BaichuanAttention层的forward中有一点用到xformers的代码：
`  

        if xops is not None and self.training:
            attn_weights = None
            # query_states = query_states.transpose(1, 2)
            # key_states = key_states.transpose(1, 2)
            # value_states = value_states.transpose(1, 2)
            # attn_output = xops.memory_efficient_attention(
            #     query_states, key_states, value_states, attn_bias=attention_mask
            # )
            with torch.backends.cuda.sdp_kernel(enable_flash=True, enable_math=True, enable_mem_efficient=True):
                attn_output = F.scaled_dot_product_attention(query_states, key_states, value_states, attn_mask = attention_mask)
            attn_output = attn_output.transpose(1, 2)
            logger.info("xops_attn:{}".format(str(attn_output)))
        else:
            attn_weights = torch.matmul(
                query_states, key_states.transpose(2, 3)
            ) / math.sqrt(self.head_dim)

            if attention_mask is not None:
                if q_len == 1:  # inference with cache
                    if len(attention_mask.size()) == 4:
                        attention_mask = attention_mask[:, :, -1:, :]
                    else:
                        attention_mask = attention_mask[:, -1:, :]
                attn_weights = attn_weights + attention_mask
                attn_weights = torch.max(
                    attn_weights, torch.tensor(torch.finfo(attn_weights.dtype).min)
                )

            attn_weights = torch.nn.functional.softmax(attn_weights, dim=-1)
            attn_output = torch.matmul(attn_weights, value_states)

            attn_output = attn_output.transpose(1, 2)
            logger.info("not_xops_attn:{}".format(str(attn_output)))`

if/else 走上面就会报错，走下面就不会。如果没有强制卸载xformers，把这里改一下强制走下面就可以。

**Comment by yajunDai** [comment]
但是按照这种方式，训练会变慢。

**Comment by ZhiningMa** [comment]
这个有不修改代码和卸载xformers的方法吗
